### PR TITLE
Add the Agent Sensor scanner

### PR DESCRIPTION
**Pull request by mrproliu**

### CODE REVIEW

File: configs/rover_configs.yaml [hunk 1]
```diff
@@ -37,30 +37,33 @@ core:
...
-  vm:
-    # Is active the VM mode to detect processes
-    active: ${ROVER_PROCESS_DISCOVERY_VM_ACTIVE:false}
+  # Scan process from linux
+  scanner:
     # The period to detect the process
-    period: ${ROVER_PROCESS_DISCOVERY_VM_PERIOD:3s}
-    finders:
+    period: ${ROVER_PROCESS_DISCOVERY_SCAN_PERIOD:3s}
+    # The process detect mode of scanner, support "REGEX", "AGENT_SENSOR"
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:44]:
```suggestion
    # The process detection modes of the scanner. Support "REGEX", "AGENT_SENSOR"
```

File: docs/en/setup/configuration/process_discovery/overview.md [hunk 2]
```diff
@@ -12,4 +12,4 @@ After the process upload is completed, the other modules could perform more oper
 ## Process Detector
 
 Process Detector is used to detect the process from the VM with the different environment:
-1. [Linux Process Detector](./linux.md)
\ No newline at end of file
+1. [Linux Process Scanner](./scanner.md)
```
Comment (wu-sheng) [inline docs/en/setup/configuration/process_discovery/overview.md:15]: What is the other options? `Container` or `Kubenetes`?

File: configs/rover_configs.yaml [hunk 3]
```diff
@@ -37,30 +37,33 @@ core:
...
+    mode: ${ROVER_PROCESS_DISCOVERY_SCAN_MODE:AGENT_SENSOR}
+    agent:
+      # Supports scanning for how long ago of the active agent
+      recent: ${ROVER_PROCESS_DISCOVERY_AGENT_RECENT:10m}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:48]: `recent` feels a little strange.

Comment (wu-sheng) [reply configs/rover_configs.yaml:48]: Why do we need this? And why the value is so long?(10m seems crazy). Rover and SDK are running in the same OS/Linux, I think there wouldn't be a timestamp gap.

File: configs/rover_configs.yaml [hunk 4]
```diff
@@ -37,30 +37,33 @@ core:
...
+    regex:
       # Use regex string to locate the processes
       # Duplicate entities cannot be reported. If multiple entity are generated, only one process will be report
       # If the multiple finders could match the same one process, only the first finder could be selected and report
-      - match_cmd_regex: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_MATCH_CMD_REGEX:}
+      - match_cmd: ${ROVER_PROCESS_DISCOVERY_REGEX_FINDER_MATCH_CMD_REGEX:}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:53]: `FINDER` or `SCANNER`, could you make them consistent in the doc and env variable?

File: docs/en/setup/configuration/process_discovery/scanner.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+# Linux Process Detector
+
+The Linux process scanner could detect any process in the Linux with finders. It has two mode:
+1. **REGEX**: could define a regex to filter which processes declare to monitor.
+2. **AGENT_SENSOR**: scan recent active agent which have process status hook.
```
Comment (wu-sheng) [inline docs/en/setup/configuration/process_discovery/scanner.md:5]: I think we need another example for this too. You need to ref to there from go2sky doc.

File: docs/en/setup/configuration/process_discovery/overview.md (hunk 2 above)
Comment (mrproliu) [reply docs/en/setup/configuration/process_discovery/overview.md:15]: Is `Kubernetes`.

File: configs/rover_configs.yaml (hunk 3 above)
Comment (mrproliu) [reply configs/rover_configs.yaml:48]: That means we only collect the process which active how minutes ago. If the process has been dead(no keepalive again), so we don't need to collect them. 

> And why the value is so long?

It's decided by how long is the process to keep alive. I could update to `1m`.

Comment (wu-sheng) [reply configs/rover_configs.yaml:48]: OK, then this name is not proper.  `period` is there for a rover, should this be `agentProcessStatusRefreshPeriod`?

`recent` has no specific meaning in this case. `recent` is describing what is happening now, not a concept of timestamp gap.

File: configs/rover_configs.yaml (hunk 4 above)
Comment (mrproliu) [reply configs/rover_configs.yaml]: Updated.

File: configs/rover_configs.yaml (hunk 3 above)
Comment (mrproliu) [reply configs/rover_configs.yaml]: Update to `process_statusr_efresh_period`, and `1m` as default.

File: docs/en/setup/configuration/process_discovery/scanner.md (hunk 5 above)
Comment (mrproliu) [reply docs/en/setup/configuration/process_discovery/scanner.md:5]: Add an example, I will create other PR to link this official documentation to go2sky.

File: configs/rover_configs.yaml (hunk 3 above)
Comment (wu-sheng) [reply configs/rover_configs.yaml]: `process_statusr_efresh_period` typo? of `statusr`?

Comment (mrproliu) [reply configs/rover_configs.yaml]: Oh, my mistake. It's `process_status_refresh_period`

File: docs/en/setup/examples/cases/agent-sensor/readme.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# Detect Process by Agent Sensor
+
+This documentation helps you to set up service which could let rover detected automatically.
+
+## Configure Service
+
+Configure your service to enable the Process Status Hook feature. Take [go2sky](https://github.com/SkyAPM/go2sky) as an example here.
```
Comment (wu-sheng) [inline docs/en/setup/examples/cases/agent-sensor/readme.md:7]: Please declare the minimum version here.

Comment (mrproliu) [reply docs/en/setup/examples/cases/agent-sensor/readme.md:7]: This feature of go2sky haven't released. 1.5.0 is fine?

Comment (wu-sheng) [reply docs/en/setup/examples/cases/agent-sensor/readme.md:7]: I think so.

Comment (mrproliu) [reply docs/en/setup/examples/cases/agent-sensor/readme.md]: updated.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**